pandas/core/generic.py

```diff
@@ -170,6 +183,11 @@ class NDFrame(PandasObject, SelectionMixin):
     _metadata = []  # type: List[str]
     _is_copy = None
     _data = None  # type: BlockManager
+    _AXIS_ALIASES = None  # type: Dict[str, int]
```

Comment:
@WillAyd if I put these in a TYPE_CHECKING block, would not need to make changes to `pandas/tests/indexing/test_indexing.py` and maintain runtime behaviour.

what's your preference here?

Reply:
because we're still supporting Python 3.5 we are using a comment-based syntax for variable annotations.

`_AXIS_ALIASES = None  # type: Dict[str, int]`

and to be valid Python syntax need to initialize to None, whereas shortly when we drop Python 3.5 we can use..

`_AXIS_ALIASES: Dict[str, int]`

and we can declare the variable type without any initialization necessary which will not change runtime behaviour.

see also https://stackoverflow.com/questions/56584140/python-3-5-type-annotated-variable-without-initial-value

so my preference is to add if TYPE_CHECKING: blocks for compatibility with Python 3.6 and simplify the adoption of Pyhton 3.6 variable annotations when we drop Python 3.5.